src/providers/okta.ts

```diff
@@ -0,0 +1,104 @@
+import { OAuth2Client } from "oslo/oauth2";
+import type { OAuth2ProviderWithPKCE } from "../index.js";
+import { TimeSpan, createDate } from "oslo";
+
+export class Okta implements OAuth2ProviderWithPKCE {
+	private client: OAuth2Client;
+	private clientSecret: string;
+
+	constructor(
+		oktaDomain: string,
+		clientId: string,
+		clientSecret: string,
+		redirectURI: string,
+		options: {
+			authServerId?: string;
+		}
+	) {
+		const authorizeEndpoint = `${oktaDomain}/oauth2/${
+			options.authServerId ? `${options.authServerId}/` : ""
+		}v1/authorize`;
+		const tokenEndpoint = `${oktaDomain}/oauth2/${
+			options.authServerId ? `${options.authServerId}/` : ""
+		}v1/token`;
+		this.client = new OAuth2Client(clientId, authorizeEndpoint, tokenEndpoint, {
+			redirectURI
+		});
+		this.clientSecret = clientSecret;
+	}
+
+	public async createAuthorizationURL(
+		state: string,
+		codeVerifier: string,
+		options?: {
+			scopes?: string[];
+		}
+	): Promise<URL> {
+		const url = await this.client.createAuthorizationURL({
+			codeVerifier,
+			scopes: [...(options?.scopes ?? []), "openid"]
```

Comment:
If a provider implements OpenID, it should always implement it (see Entra, Google, etc)

Reply:
I'm a bit confused it's being added separate from the passed scopes to always be included?

Reply:
See the Google provider. `openid` scope is always included

Reply:
Yes, I'm always including it too here. I'm taking the passed scopes and spreading them (or an empty array if undefined) inside another array with `openid` already in it. The outcome will be that no matter what scopes are passed `openid` will always be included.

Reply:
Oh wait oof lol - my bad